### PR TITLE
[API keys] Add full_name and email to API key doc and use them to populate authing User

### PR DESCRIPTION
**Pull request by lloydmeta**
The API key document currently doesn't include the user's full_name or email attributes, and as a result, when those attributes return `null` when hitting `GET`ing  `/_security/_authenticate`, and in the SAML response from the [IdP Plugin](https://github.com/elastic/elasticsearch/pull/54046).

This changeset adds those fields to the document and extracts them to fill in the User when authenticating. They're effectively going to be a snapshot of the User from when the key was created, but this is in line with roles and metadata as well.

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java
Comment (jkakavas) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java:211]: nit: use Bruce Baner, or anything else for that matter so that it's easier to spot it's different from the principal, in the eventual case of troubleshooting a test failure

**Review by jkakavas — APPROVED** [review]
LGTM! Thanks LLoyd. I consider this a bug so I agree it should go in the next patch releases ( 7.8.2, 7.9.1 ) too.  I don't see this breaking a client's response, the Authenticate API is supposed to be returning these two fields either way

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java
Comment (lloydmeta) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java:211]: ++ nice one, fixed.

**Comment by jkakavas** [comment]
@lloydmeta we usually tend to wait for a review from all reviewers that we ask. No big deal though, Tim can always comment and we can adjust if he has any suggestions/objections. 

I realize this is not the practice in all repos/teams ,just bringing it up for future reference:)

**Comment by lloydmeta** [comment]
> @lloydmeta we usually tend to wait for a review from all reviewers that we ask. No big deal though, Tim can always comment and we can adjust if he has any suggestions/objections.
> 
> I realize this is not the practice in all repos/teams ,just bringing it up for future reference:)

Oh, sorry about that. Thanks for letting me know :) makes sense to me. Will keep that in mind.

**Review by tvernum — COMMENTED** [review]
LGTM

**Comment by tvernum** [comment]
I dropped `7.8.2`. Now that `7.9.0` is released, we don't maintain 7.8 and we don't plan to release a `7.8.2`
Backporting to 7.8 is a waste of effort.

@lloydmeta Do you want us to backport this, or do you have it under control?

**Comment by lloydmeta** [comment]
Thanks @tvernum I'm happy to handle the backporting as well.

**Comment by lloydmeta** [comment]
Backports:
- To 7.10 (7.x branch) https://github.com/elastic/elasticsearch/pull/61403
- To 7.9 (7.9 branch) https://github.com/elastic/elasticsearch/pull/61404

Would appreciate a spot check from one of you to make sure I did those right before merging @jkakavas @tvernum

**Pull request merged**
